lib/improver/tests/argparser/test_ArgParser.py

```diff
@@ -0,0 +1,328 @@
+            args = parser.parse_args()
+            args = vars(args).keys()
+            self.assertItemsEqual(args, ['foo', 'bar'])
+
+    def test_create_argparser_all_arguments(self):
+        """Test that creating an ArgParser with compulsory, centralized and
+        specific arguments adds the arguments from all 3 collections."""
+
+        compulsory_arguments = {'foo': (['--foo'], {})}
+        centralized_arguments = {'bar': (['--bar'], {})}
+        specific_arguments = [(['--baz'], {})]
+
+        with patch('improver.argparser.ArgParser.COMPULSORY_ARGUMENTS',
+                   compulsory_arguments):
+            with patch('improver.argparser.ArgParser.CENTRALIZED_ARGUMENTS',
+                       centralized_arguments):
+                parser = ArgParser(central_arguments=['bar'],
+                                   specific_arguments=specific_arguments)
+                args = parser.parse_args()
+                args = vars(args).keys()
+                self.assertItemsEqual(args, ['foo', 'bar', 'baz'])
+
+
+class Test_add_arguments(QuietTestCase):
+
+    """Test the add_arguments method."""
+
+    def test_adding_multiple_arguments(self):
+        """Test that we can successfully add multiple arguments to the
+        ArgParser."""
+
+        # we will not actually pass anything in, so the Namespace will receive
+        # the defaults (if any) - only check the keys of the Namespace derived
+        # dictionary
+        args_to_add = [(['--foo'], {}),
+                       (['--bar', '--b'], {})]
+
+        expected_namespace_keys = ['foo', 'bar']  # + compulsory...
+
+        # explicitly pass nothing in - will only have compulsory arguments
+        # and the ones we added...
+        parser = ArgParser(central_arguments=None,
+                           specific_arguments=None)
+
+        parser.add_arguments(args_to_add)
+        result_args = parser.parse_args()
+        result_args = vars(result_args).keys()
+        # we could also add compulsory arguments to expected_namespace_keys
+        # and then assertItemsEqual - (order unimportant), but this
+        # is unnecessary - just use loop:
+        # (or we could patch compulsory arguments to be an empty dictionary)
+        for expected_arg in expected_namespace_keys:
+            self.assertIn(expected_arg, result_args)
+
+    def test_adding_argument_with_defined_kwargs_dict(self):
+        """Test that we can successfully add an argument to the ArgParser,
+        when the argspec contained kwargs."""
+
+        # length of argspec is 2...
+        args_to_add = [(['--foo'], {'action': 'store_true'})]
+        expected_arg = 'foo'
+
+        parser = ArgParser(central_arguments=None,
+                           specific_arguments=None)
+
+        parser.add_arguments(args_to_add)
+        result_args = parser.parse_args()
+        result_args = vars(result_args).keys()
+        self.assertIn(expected_arg, result_args)
```

Comment:
This doesn't actually check it does anything with the dictionary, you are only checking something with the flags. Is there a way to check the store true has been used? Is it important?